lib/index.js

```diff
@@ -62,6 +66,21 @@ function start(startingConfig) {
   };
 }
 
+function setFlushOnExit() {
+  function handleBeforeExit(exitCode) {
+    debug('flushing last beacons before exiting');
+    transmitter.handlePeriodicTasks()
+      .then(() => {
+        process.exit(exitCode);
+      })
+      .catch(() => {
+        process.exit(exitCode);
+      });
+  }
+
+  process.on('beforeExit', handleBeforeExit);
```

Comment:
Why not SIGTERM? Please look at https://stackoverflow.com/questions/32457471/listen-for-beforeexit-or-sigterm-to-trigger-cleanup-in-node, I think SIGTERM is what you want

Reply:
SIGTERM almost certainly isn't the thing we care about, as that's not the only reason the some unknown node process would exit

Reply:
@shaimendel that article is Heroku specific, it's not universally relevant?

Reply:
Yup, I was confused with what we did in homebase, where you also always get SIGTERM (thanks to kubernetes)

Reply:
It seems that `beforeExit` only happens when the program ends gracefully, so I assume it isn't called when unhandled exceptions occur for example (or any other non-graceful flow).
This means that we will flush only when the tests end gracefully, which is reasonable

Reply:
Yeah, this is a bit of a weird one, as its possible that in some "bad exit" cases we might want to flush. 

We might need in future to expose a way to flush explicitly via the API, for anyone with "weird" requirements, so that at least users can take control of this where needed. A bit like we do in our registry tests with our Redis and Postgres connection pools